### PR TITLE
FLUID-6755: Update to last released version of fluid-lint-all.

### PR DESCRIPTION
**Pull request by duhrer**
In recent conversations with @jobara, we discovered that there linting errors reported with any newer version of `fluid-lint-all` than is currently used in master.  These are related to a refactor in the checks for emphasis and strong emphasis, which are now able to detect the fact that we use double asterisks for some "strong emphasis" blocks, and double underscores for others.

### CODE REVIEW

**Comment by duhrer** [comment]
I also tested with the upcoming 1.2.3 release of fluid-lint-all, which reveals that the jsdoc linting now reports errors around functions which include a `@returns` block in their jsdocs, but which do not always return anything:

```
14:27:46.862:  Errors returned by eslint.js:
14:27:46.863:  
14:27:46.863:    - src/framework/core/js/Fluid.js:
14:27:46.863:      (1276:1) JSDoc @return declaration present but return expression not available in function. (jsdoc/require-returns-check)
14:27:46.863:  
14:27:46.863:    - src/framework/core/js/FluidIoC.js:
14:27:46.863:      (352:1) JSDoc @return declaration present but return expression not available in function. (jsdoc/require-returns-check)
14:27:46.864:      (2228:1) JSDoc @return declaration present but return expression not available in function. (jsdoc/require-returns-check)
14:27:46.864:  
14:27:46.864:    - src/framework/core/js/FluidView.js:
14:27:46.864:      (573:1) JSDoc @return declaration present but return expression not available in function. (jsdoc/require-returns-check)
14:27:46.864:  
14:27:46.864:    - src/framework/core/js/MessageResolver.js:
14:27:46.864:      (112:1) JSDoc @return declaration present but return expression not available in function. (jsdoc/require-returns-check)
14:27:46.864:  
14:27:46.864:    - src/framework/preferences/js/PrimaryBuilder.js:
14:27:46.864:      (117:1) JSDoc @return declaration present but return expression not available in function. (jsdoc/require-returns-check)
14:27:46.864:  
14:27:46.864:    - src/framework/preferences/js/Store.js:
14:27:46.864:      (128:1) JSDoc @return declaration present but return expression not available in function. (jsdoc/require-returns-check)
```

As an example, [the function flagged in `MessageResolver`](https://github.com/fluid-project/infusion/blob/a0baa7530568abb68a158c00a740b29464d0d50a/src/framework/core/js/MessageResolver.js#L123) indicates that a function or string might be returned, but does not mention that there might be no return.  Adding `undefined` to the ORed list of return types resolves this.

**Comment by duhrer** [comment]
I have updated this pull to fix those linting errors, which were generally correct, in that we often said we would return something but defaulted to `undefined` in a few cases.  I added "final" returns with a sensible default in a few cases.  In others I added `undefined` to the return definition, with an explanation.

There was one instance where the function didn't return because it threw via `fluid.fail`, I disabled the check for that function.

File: src/framework/core/js/Fluid.js [hunk 1]
```diff
@@ -1273,13 +1273,15 @@ fluid.bind = function (obj, fnName, args) {
...
 fluid.proxyComponentArgs = fluid.identity;
+/* eslint-disable jsdoc/require-returns-check */
 /**
  * Allows for the calling of a function from an EL expression "functionPath", with the arguments "args", scoped to an framework version "environment".
  * @param {Object} functionPath - An EL expression
  * @param {Object} args - An array of arguments to be applied to the function, specified in functionPath
  * @param {Object} [environment] - (optional) The object to scope the functionPath to  (typically the framework root for version control)
  * @return {Any} - The return value from the invoked function.
  */
+/* eslint-enable jsdoc/require-returns-check */
```
Comment (jobara) [inline src/framework/core/js/Fluid.js:1284]: @the-t-in-rtf this is because the return is in the conditional?

**Comment by jobara** [comment]
@duhrer the changes look good. Not sure if @amb26 wants to look through these as well. I guess we should also wait for the stable release of fluid-lint-all before merging this.

**Comment by amb26** [comment]
Thanks all, I will be able to take a look at this tomorrow

**Comment by duhrer** [comment]
> @duhrer the changes look good. Not sure if @amb26 wants to look through these as well. I guess we should also wait for the stable release of fluid-lint-all before merging this.

Yes, that was my idea, verify that we're happy with the changes we have to make with the code (or make changes to the linting config), then merge this one with the released version of `fluid-lint-all`.

**Pull request merged**